pyodide-build/pyodide_build/common.py

```diff
@@ -209,3 +171,30 @@ def get_make_environment_vars():
             value = value.strip("'").strip()
             environment[varname] = value
     return environment
+
+
+def search_pyodide_root(base: str | Path, depth: int, max_depth: int = 5) -> Path:
+    """
+    Recursively search for the root of the Pyodide repository,
+    by looking for the pyproject.toml file in the parent directories
+    which contains [tool.pyodide] section.
+    """
+
+    if depth >= max_depth:
+        raise FileNotFoundError(
+            "Could not find Pyodide root directory. If you are not in the Pyodide directory, set `PYODIDE_ROOT=<pyodide-root-directory>`."
+        )
+
+    base = Path(base)
```

Comment:
Wait you need to use depth somewhere. And also check that we are not walking higher than '/'. Something like,
```py
    # at the top
    base = Path(base)
    max_depth = min(max_depth, len(base.parents))
	if depth >= max_depth:
    ...
    
    current_dir = base.parents[depth]
    pyproject_file = current_dir / "pyproject.toml"
```
?

Reply:
I guess `Path("/").parent == Path("/")` so not checking that case isn't **wrong** persay it's just wasteful.